src/corona_hakab_model/sub_matrices.py

```diff
@@ -0,0 +1,26 @@
+from scipy.stats import rv_discrete
+from agent import Agent
+from typing import List
+
+
+class CircularConnectionsMatrix:
+
+    __slots__ = "type", "agents", "circle_size_probability", "connection_strength"
+
+    # todo switch type to a named-tuple
+    def __init__(self, type: str, agents: List[Agent], circle_size_probability: rv_discrete, connection_strength: float):
+        self.type = type
+        self.agents = agents
+        self.circle_size_probability = circle_size_probability
+        self. connection_strength = connection_strength
+
+class NonCircularConnectionMatrix:
```

Comment:
maybe they should have a common super-type?

Reply:
possible, but i don't think it really matters, at least not for the upcomming version